lib/vagrant-google/action/read_ssh_info.rb

```diff
@@ -25,12 +25,17 @@ def initialize(app, env)
         end
 
         def call(env)
-          env[:machine_ssh_info] = read_ssh_info(env[:google_compute], env[:machine])
+          # Get the zone we're going to use
+          zone = env[:machine].provider_config.zone
+          # Check if we should use public or private IP address of the instance
+          use_private_ip = env[:machine].provider_config.get_zone_config(zone).use_private_ip
+
+          env[:machine_ssh_info] = read_ssh_info(env[:google_compute], env[:machine], use_private_ip)
 
           @app.call(env)
         end
 
-        def read_ssh_info(google, machine)
+        def read_ssh_info(google, machine, use_private_ip)
```

Comment:
In my opinion, it's better to DRY this up and leave the call method unchanged, instead of adding another argument, since we're already getting the zone info here. Something like:
```
          # Find the machine and get the private ip config
          zone = machine.provider_config.zone
          server = google.servers.get(machine.id, zone)
          use_private_ip = machine.provider_config.get_zone_config(zone).use_private_ip
```

What do you think?